demo/src/package-lock-generator.ts

```diff
@@ -0,0 +1,241 @@
+import {Firestore} from '@google-cloud/firestore';
+import child_process from 'child_process';
+import firebaseAdmin from 'firebase-admin';
+import * as fsExtra from 'fs-extra';
+import os from 'os';
+import * as path from 'path';
+import url from 'url';
+import {promisify} from 'util';
+import zlib from 'zlib';
+
+import {Cache} from './cache';
+import {fetch} from './util';
+
+const MEMORY_TOTAL_MB = 1024;
+const MEMORY_RESERVED_MB = 128;
+
+const exec = promisify(child_process.exec);
+
+/**
+ * Basic representation of the contents of package-lock.json file.
+ */
+type PackageDefinition = {
+  dependencies?: {[key: string]: PackageDefinition};
+  version?: string;
+};
+
+/**
+ * Flat representation of a package-lock file.
+ */
+export type PackageVersionMap = {
```

Comment:
I think I prefer the name PackageVersionMap since its a representation of package -> version. The fact its generated from package locks is perhaps more incidental.